specification/hugr.md

```diff
@@ -1063,7 +1063,46 @@ in {a}\* (i.e. there is no hierarchy relation between them).
 
 There are the following primitive operations.
 
-##### Replacement method
+##### Replacement methods
+
+###### `SimpleReplace`
+
+This method is used for simple replacement of dataflow subgraphs consisting of
+leaf nodes.
+
+Given a DFG-convex set $S$ of IDs of leaf nodes in a DSG, let:
+
+  - $\textrm{inp}(S)$ be the set of input ports of nodes in $S$ whose source is
+    in $\Gamma \setminus S$;
+  - $\textrm{out}(S)$ be the set of input ports of nodes in $\Gamma \setminus S$
+    whose source is in $S$.
+
+Given a DFG node $P$, let:
+
+  - $\textrm{inp}(P)$ be the set of successor ports of edges from output ports
+    of the Input node of $P$;
+  - $\textrm{out}(P)$ be the set of input ports of the Output node of $P$.
+
+The method takes as input:
+
+  - the ID of a DFG node $P$ in $\Gamma$;
+  - a DFG-convex set $S$ of IDs of leaf nodes that are children of $P$ (not
+    including the Input and Output nodes);
+  - a hugr whose root is a DFG node $R$ with only leaf nodes as children;
+  - a map $\nu_\textrm{inp}: \textrm{inp}(R) \to \textrm{inp}(S)$;
+  - a map $\nu_\textrm{out}: \textrm{out}(S) \to \textrm{out}(R)$.
+  
+The new hugr is then derived by:
+  
+  - adding all children of $R$, and all edges between them, to $\Gamma$;
```

Comment:
I think you do not want to add the new `Input` and `Output` nodes from $R$

Reply:
Oh, sorry, you remove these at the end, but maybe it makes sense just not to add them

Reply:
Yes, in fact in the implementation I didn't add them. I will update the spec. (Just have to be careful about distinguishing the nodes of R from the newly-added copies.)

Reply:
Done in https://github.com/CQCL-DEV/hugr/pull/85/commits/7d1535e1197dace3027fc359b24f0107c9c996c9 .